### PR TITLE
feat:adds logic for copying additional resources to the build directory

### PR DESCRIPTION
**Pull request by g-awmalik**

### CODE REVIEW

File: infra/blueprint-test/pkg/krmt/krm.go [hunk 1]
```diff
@@ -58,6 +59,12 @@ func WithDir(dir string) krmtOption {
 	}
 }
 
+func WithResourcesDir(rscsDir string) krmtOption {
```
Comment (bharathkkb) [inline infra/blueprint-test/pkg/krmt/krm.go:62]: Should we do `WithAdditionalResources(res ...string)` instead? `WithAdditionalResources` name implies these are additional resources and  `res` can be a file or dir and multiple res can be passed in.

Comment (g-awmalik) [reply infra/blueprint-test/pkg/krmt/krm.go]: Makes sense.

File: infra/blueprint-test/pkg/krmt/krm.go [hunk 2]
```diff
@@ -58,6 +59,14 @@ func WithDir(dir string) krmtOption {
 	}
 }
 
+func WithAdditionalResources(rscsDir ...string) krmtOption {
```
[the remark this answers is not in the document]
Comment (g-awmalik) [reply infra/blueprint-test/pkg/krmt/krm.go:62]: Acked

File: infra/blueprint-test/pkg/krmt/krm.go [hunk 3]
```diff
@@ -58,6 +59,14 @@ func WithDir(dir string) krmtOption {
...
+	return func(f *KRMBlueprintTest) {
+		for _, dir := range rscsDir {
+			f.resources = append(f.resources, dir)
```
[the remark this answers is not in the document]
Comment (g-awmalik) [reply infra/blueprint-test/pkg/krmt/krm.go:65]: We could but I wanted to keep this method clean and validate stuff when we create the test. Looks and reads better that way.

File: infra/blueprint-test/pkg/krmt/krm.go [hunk 4]
```diff
@@ -30,6 +30,7 @@ type KRMBlueprintTest struct {
 	discovery.BlueprintTestConfig                          // additional blueprint test configs
 	name                          string                   // descriptive name for the test
 	exampleDir                    string                   // directory containing KRM blueprint example
+	resources                     []string                 // paths to directories or files containting additional resources to be applied
```
[the remark this answers is not in the document]
Comment (g-awmalik) [reply infra/blueprint-test/pkg/krmt/krm.go]: Acked

File: infra/blueprint-test/pkg/krmt/krm.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (bharathkkb) [reply infra/blueprint-test/pkg/krmt/krm.go]: makes sense

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**